### DISCUSSION TITLE
Lazer Collections

### DISCUSSION THREAD

**Discussion by AriizuCA** [ppy/osu#22068] [Ideas] [answered]
Version 2022.1228.0-lazer

Collections are handled in osu!lazer completely differently compared to the stable client.  As someone who basically lives in the sorted collections tab, the switch over to lazer has been tedious for me to navigate between collections.  If I want to jump between collections quickly, I have to click on a drop-down menu and then find the collection I want.

My issue is that I have over 300 collections so having to either use the scroll wheel or scroll bar (which is a tiny dot with how many collections I have) makes the task extremely difficult to do quickly compared to the stable client.  This drop-down menu also does not allow right-click dragging as the stable client does, so I also lose the muscle memory of locating collections through that feature.

If the collection dropdown menu is here to stay in its current state, then I think what could help alleviate some of the annoyance of scrolling through a large collection library is to include a search box that can allow the user to isolate specific collections.

**Comment by peppy** [comment, marked as answer]
What is right click dragging in a dropdown?

**Reply by Kaerulans** [reply]
I think what's meant is stable's way of scrolling through collections or songs in song select by holding right click. You can scroll through songs in lazer that way as well, but since collections can no longer be viewed as entries in song select, browsing collections is now much more cumbersome

**Reply by AriizuCA** [reply]
Right-click dragging is what I use when I need to quickly scroll through a large list of beatmaps or collections.  This feature exists within osu!lazer similar to stable but only within the beatmap selection area.  Below is a clip from the stable client where I am right-click dragging within the collection grouping.  One of the nice benefits of right-clicking within the song select screen is that you can also just right-click anywhere and you will be sent to that exact location (this is shown a couple of times within the first video near the end).  This allows you to get to the top, bottom, middle, where ever you want quickly, particularly useful if you specifically place collection groups near those positions where you want to constantly access.
https://user-images.githubusercontent.com/54725714/211218868-6251b237-5d22-4d95-b941-bd89a89b2865.mp4

This second video shows how right-click dragging is enabled within the song select screen of osu!lazer but not within the collection drop-down menu.  Right-click dragging does nothing within that menu and the only way to currently navigate through it is to either use your mouse scroll wheel (which will take forever if you have a large collection) or click and drag the scroll bar.  Note that this window is much smaller so moving through collections with the scroll bar requires finer control compared to the current navigation of collections in the stable client.
https://user-images.githubusercontent.com/54725714/211218873-6e34f818-690d-4305-94a0-1e7f0f7a73be.mp4

This final video is just to showcase that there seems to be an invisible line drawn for the song select area that right-click dragging can only work within.  I think the bounding box should extend to the scoreboard/map details area edge.
https://user-images.githubusercontent.com/54725714/211218876-7bc58f3d-1a58-4f74-9eaa-3e0c736f7129.mp4

Overall, if collections remain in the drop-down menu style, there needs to be a way for players to quickly access a large library if they are utilizing the collection feature to its fullest potential.  Giving access to a search function within the collections menu will allow dedicated collection players a quick way to navigate through their collections.  This, I believe would be better than right-click dragging anyway since it would cut down on time spent searching for a specific collection as long as you name the collections appropriately.

**Reply by peppy** [reply]
Okay, so this whole request is just another one for grouping options.

They are coming in the future. Please wait for that.

**Reply by AriizuCA** [reply]
I'm suggesting putting a search bar within the dropdown collection menu, highlighted in orange below.
![searchbar](https://user-images.githubusercontent.com/54725714/211250118-6b3a871e-d048-45f0-8e4c-ab75629a1806.png)

**Reply by peppy** [reply]
All dropdowns are going to be searchable in the future. See #21784.
